### DISCUSSION TITLE
force-cube: NODATA not found...

### DISCUSSION THREAD

**Discussion by JonasardoLundUniv** [davidfrantz/force#279] [Q&A]
Hi

I can't cube a Sentinel-2 L2A data set (not generated by FORCE but downloaded from gs)  due to some NoData issue, trying:
`<force-suite force-cube -r near -s 10 -t Int16  -o ./cubed/ ./T38PNQ/*/MTD*.xml>` (or similar)   yields:

force-cube: NODATA not found in /projects/eko/fs2/FORCE/RAMONA/ETHIOPIA/T38PNQ/S2A_MSIL2A_20181214T072311_N0211_R006_T38PNQ_20181214T113540.SAFE/MTD_MSIL2A.xml
  
Even if I set no data with gdal_edit.py for all MTD_MSIL2A.xml files (and all *jp2 files). No difference.
  MTD_MSIL2A.xml don't seem to be updated by gdal_edit.py (NoDataValue not found).
  *.jp2 files get <NoDataValue>-9.99900000000000E+03</NoDataValue> in *.jp2.aux.xml

File not changed by gdal when trying to set NoData:
[jonasardo@aurora-nateko2 ETHIOPIA]$ date
Wed Mar 15 13:13:06 CET 2023
[jonasardo@aurora-nateko2 ETHIOPIA]$ ls -al  /projects/eko/fs2/FORCE/RAMONA/ETHIOPIA/T38PNQ/S2A_MSIL2A_20181214T072311_N0211_R006_T38PNQ_20181214T113540.SAFE/MTD_MSIL2A.xml
-rwxr--r-- 1 jonasardo leo 53275 Mar  8 22:47 /projects/eko/fs2/FORCE/RAMONA/ETHIOPIA/T38PNQ/S2A_MSIL2A_20181214T072311_N0211_R006_T38PNQ_20181214T113540.SAFE/MTD_MSIL2A.xml
[jonasardo@aurora-nateko2 ETHIOPIA]$ `gdal_edit.py -a_nodata -9999  /projects/eko/fs2/FORCE/RAMONA/ETHIOPIA/T38PNQ/S2A_MSIL2A_20181214T072311_N0211_R006_T38PNQ_20181214T113540.SAFE/MTD_MSIL2A.xml`
[jonasardo@aurora-nateko2 ETHIOPIA]$ ls -al  /projects/eko/fs2/FORCE/RAMONA/ETHIOPIA/T38PNQ/S2A_MSIL2A_20181214T072311_N0211_R006_T38PNQ_20181214T113540.SAFE/MTD_MSIL2A.xml
-rwxr--r-- 1 jonasardo leo 53275 Mar  8 22:47 /projects/eko/fs2/FORCE/RAMONA/ETHIOPIA/T38PNQ/S2A_MSIL2A_20181214T072311_N0211_R006_T38PNQ_20181214T113540.SAFE/MTD_MSIL2A.xml
[jonasardo@aurora-nateko2 ETHIOPIA]$

Perhaps more a GDAL-question than a FORCE-question, but still.
Any suggestions? Using gdal 3.5.2 on LINUX (CentOS 7)

Regards
/Jonas Ardö

**Comment by davidfrantz** [comment]
Hi Jonas,

the xml files only contain metadata and cannot be cubed. I guess the error message is a bit misleading in that regard.

Best,
David
